src/app/components/form/DualCurrencyField/index.tsx

```diff
@@ -1,22 +1,37 @@
-import { useEffect, useRef } from "react";
+import { useCallback, useEffect, useRef, useState } from "react";
 import { useTranslation } from "react-i18next";
+import { useAccount } from "~/app/context/AccountContext";
+import { useSettings } from "~/app/context/SettingsContext";
 import { classNames } from "~/app/utils";
-
 import { RangeLabel } from "./rangeLabel";
 
+export type DualCurrencyFieldChangeEventTarget = HTMLInputElement & {
+  valueInFiat: number;
+  formattedValueInFiat: string;
+  valueInSats: number;
+  formattedValueInSats: string;
+};
+
+export type DualCurrencyFieldChangeEvent =
+  React.ChangeEvent<HTMLInputElement> & {
+    target: DualCurrencyFieldChangeEventTarget;
+  };
+
 export type Props = {
   suffix?: string;
   endAdornment?: React.ReactNode;
-  fiatValue: string;
   label: string;
   hint?: string;
   amountExceeded?: boolean;
   rangeExceeded?: boolean;
+  baseToAltRate?: number;
+  showFiat?: boolean;
+  onChange?: (e: DualCurrencyFieldChangeEvent) => void;
 };
 
 export default function DualCurrencyField({
```

Comment:
Now its a good enough complexity. maybe add comments wherere necessary why such and such thing is used

Reply:
I've added some comments to the code and removed baseToAltRate since it was dead code from one of the previous iterations :+1: